### ISSUE TITLE
Circular dependency in current version's gem-spec

### ISSUE THREAD

**Issue by klausmeyer** [kevinzen/send_nsca#6]
Hi,

the current version of the gem cannot be installed using bundler:

```ruby
source 'https://rubygems.org'

gem 'send_nsca'
```

```
» bundle                                                                                                                                                                                                                                                          meyer@kmdev
Fetching gem metadata from https://rubygems.org/...........
Fetching version metadata from https://rubygems.org/...
Fetching dependency metadata from https://rubygems.org/..
Resolving dependencies...
Your Gemfile requires gems that depend on each other, creating an infinite loop. Please remove gem 'send_nsca' and try again.
```

I think this is because the gem is requiring itself in the [gem-spec](
https://github.com/kevinzen/send_nsca/blob/f7d962ad22796389ccbbe4a4af17b46a0817fb33/send_nsca.gemspec#L41)

Here my setup:

```
» ruby --version                                                                                                                                                                                                                                                  meyer@kmdev
ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin14]
--- 9:32:14 ---                                                                                                                                                                                                                                                               
~  ruby-2.2.2
» gem --version                                                                                                                                                                                                                                                   meyer@kmdev
2.4.6
--- 9:32:19 ---                                                                                                                                                                                                                                                               
~  ruby-2.2.2
» bundle --version                                                                                                                                                                                                                                                meyer@kmdev
Bundler version 1.10.5
--- 9:32:24 ---                                                                                                                                                                                                                                                               
~  ruby-2.2.2
```
Best, Klaus

**Comment by Kyrremann** [comment]
Hum...that sounds strange. I tried to create a Gemfile with just
```ruby
source 'https://rubygems.org'

gem 'send_nsca'
```

And it installed nicely for me.
```
[14:10:40]send_nsca_test$ bundle install
Fetching gem metadata from https://rubygems.org/...........
Fetching additional metadata from https://rubygems.org/..
Resolving dependencies...
Using rake 10.4.2
Installing addressable 2.3.8
Using builder 3.2.2
Using bundler 1.7.12
Installing thread_safe 0.3.5
Using descendants_tracker 0.0.4
Using multipart-post 2.0.0
Using faraday 0.9.1
Using git 1.2.9.1
Installing hashie 3.4.2
Installing multi_json 1.11.1
Using mini_portile 0.6.2
Using nokogiri 1.6.6.2
Installing jwt 1.5.1
Using multi_xml 0.5.5
Using rack 1.6.4
Using oauth2 1.0.0
Installing github_api 0.12.3
Installing highline 1.7.2
Using rdoc 4.2.0
Using jeweler 2.0.1
Installing send_nsca 0.6.0
Your bundle is complete!
Use `bundle show [gemname]` to see where a bundled gem is installed.
[14:11:05]send_nsca_test$ ruby --version
ruby 2.2.0p0 (2014-12-25 revision 49005) [x86_64-linux]
[14:11:15]send_nsca_test$ gem --version
2.4.5
[14:11:22]send_nsca_test$ bundle --version
Bundler version 1.7.12
```

I've never used Jewler before, I just took over maintenance since I needed some new features. So it may been a configuration issue. I'll give it a new try with the same ruby, gem, and bundler versions.

**Comment by Kyrremann** [comment]
I manage to reproduce with the latest Ruby, Gem, and Bundler version.
I'll see if I can manage a fix tonight.

**Comment by klausmeyer** [comment]
Thank you :+1:

**Comment by Kyrremann** [comment]
Sorry, but haven't been able to create a good fix yet. I'm working on removing the jeweler-gem, as I don't see the need for it. But I don't have much time to work on this. I'll let you know when I've gotten any further.

**Comment by klausmeyer** [comment]
Hi, that's not a problem. I simply use an older version of the gem which is working fine at the moment.

**Comment by elimydlarz** [comment]
You can avoid this problem by using version 0.5.0. Update your `Gemfile` like so: `gem 'send_nsca', '0.5.0'`

**Comment by Kyrremann** [comment]
Finally got some time to fix the issue, see #9.

**Comment by klausmeyer** [comment]
Hi,

thanks for fixing the issue. I can confirm that now the gem can be installed and used in version `0.6.1` in my setup.

But maybe you should think about making the jeweler thingy a [development_dependency][1] because now the following gems would be added in my project when I upgrade `send_nsca`:

```
diff --git a/Gemfile.lock b/Gemfile.lock
index 213fde9..7040900 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -63,6 +63,8 @@ GEM
       execjs
     coffee-script-source (1.10.0)
     concurrent-ruby (1.0.1)
+    descendants_tracker (0.0.4)
+      thread_safe (~> 0.3, >= 0.3.1)
     diff-lcs (1.2.5)
     docile (1.1.5)
     dotenv (2.0.2)
@@ -78,8 +80,18 @@ GEM
       railties (>= 3.0.0)
     faraday (0.9.2)
       multipart-post (>= 1.2, < 3)
+    git (1.3.0)
+    github_api (0.13.1)
+      addressable (~> 2.4.0)
+      descendants_tracker (~> 0.0.4)
+      faraday (~> 0.8, < 0.10)
+      hashie (>= 3.4)
+      multi_json (>= 1.7.5, < 2.0)
+      oauth2
     globalid (0.3.6)
       activesupport (>= 4.1.0)
+    hashie (3.4.3)
+    highline (1.7.8)
     hipchat (1.5.2)
       httparty
       mimemagic
@@ -87,11 +99,21 @@ GEM
       json (~> 1.8)
       multi_xml (>= 0.5.2)
     i18n (0.7.0)
+    jeweler (2.0.1)
+      builder
+      bundler (>= 1.0)
+      git (>= 1.2.5)
+      github_api
+      highline (>= 1.6.15)
+      nokogiri (>= 1.5.10)
+      rake
+      rdoc
     jquery-rails (4.0.5)
       rails-dom-testing (~> 1.0)
       railties (>= 4.2.0)
       thor (>= 0.14, < 2.0)
     json (1.8.3)
+    jwt (1.5.1)
     launchy (2.4.3)
       addressable (~> 2.3)
     letter_opener (1.4.1)
@@ -113,6 +135,12 @@ GEM
     mysql2 (0.4.2)
     nokogiri (1.6.7.2)
       mini_portile2 (~> 2.0.0.rc2)
+    oauth2 (1.1.0)
+      faraday (>= 0.8, < 0.10)
+      jwt (~> 1.0, < 1.5.2)
+      multi_json (~> 1.3)
+      multi_xml (~> 0.5)
+      rack (>= 1.2, < 3)
     parse-cron (0.1.4)
     puma (2.16.0)
     rack (1.6.4)
@@ -144,7 +172,13 @@ GEM
       rake (>= 0.8.7)
       thor (>= 0.18.1, < 2.0)
     rake (10.5.0)
+    rdoc (4.2.2)
+      json (~> 1.4)
     ref (2.0.0)
+    rspec (3.4.0)
+      rspec-core (~> 3.4.0)
+      rspec-expectations (~> 3.4.0)
+      rspec-mocks (~> 3.4.0)
     rspec-activemodel-mocks (1.0.2)
       activemodel (>= 3.0)
       activesupport (>= 3.0)
@@ -175,7 +209,9 @@ GEM
       sprockets (>= 2.8, < 4.0)
       sprockets-rails (>= 2.0, < 4.0)
       tilt (>= 1.1, < 3)
-    send_nsca (0.5.0)
+    send_nsca (0.6.1)
+      jeweler (~> 2.0, >= 2.0.1)
+      rspec (~> 3.4)
     shoulda-matchers (3.0.1)
       activesupport (>= 4.0.0)
     simplecov (0.11.1)
@@ -236,7 +272,7 @@ DEPENDENCIES
   rspec-collection_matchers
   rspec-rails
   sass-rails
-  send_nsca (~> 0.5.0)
+  send_nsca
   shoulda-matchers
   simplecov
   simplecov-rcov
```

Best Klaus

[1]: http://guides.rubygems.org/specification-reference/#add_development_dependency

**Comment by Kyrremann** [comment]
Yeah, that sounds like a good idé. Hopefully it should be an easy fix.

**Comment by Kyrremann** [comment]
Fixed in v0.6.2

**Comment by klausmeyer** [comment]
Prefect, thank you :sunglasses: 

```
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -175,7 +175,7 @@ GEM
       sprockets (>= 2.8, < 4.0)
       sprockets-rails (>= 2.0, < 4.0)
       tilt (>= 1.1, < 3)
-    send_nsca (0.5.0)
+    send_nsca (0.6.2)
     shoulda-matchers (3.0.1)
       activesupport (>= 4.0.0)
     simplecov (0.11.1)
@@ -236,7 +236,7 @@ DEPENDENCIES
   rspec-collection_matchers
   rspec-rails
   sass-rails
-  send_nsca (~> 0.5.0)
+  send_nsca
   shoulda-matchers
   simplecov
   simplecov-rcov
```
